### PR TITLE
Add script to find orphaned cables

### PR DESCRIPTION
**Pull request by DanSheps**
Pretty basic, query for orphaned or partially orphaned cables and return the list.

### CODE REVIEW

**Comment by kkthxbye-code** [comment]
One-liner if interested:

```python
from django.db.models import Count

Cable.objects.annotate(
    aterm=Count('terminations', filter=Q(terminations__cable_end="A")),
    bterm=Count('terminations', filter=Q(terminations__cable_end="B")),
).filter(Q(aterm=1, bterm=0) | Q(aterm=0, bterm=1) | Q(aterm=0, bterm=0))
```

**Comment by kkthxbye-code** [comment]
Slight edit, added `__gte`:

```python
Cable.objects.annotate(
    aterm=Count('terminations', filter=Q(terminations__cable_end="A")),
    bterm=Count('terminations', filter=Q(terminations__cable_end="B")),
).filter(Q(aterm__gte=1, bterm=0) | Q(aterm=0, bterm__gte=1) | Q(aterm=0, bterm=0))
```

**Comment by DanSheps** [comment]
> Slight edit, added `__gte`:
> 
> ```python
> Cable.objects.annotate(
>     aterm=Count('terminations', filter=Q(terminations__cable_end="A")),
>     bterm=Count('terminations', filter=Q(terminations__cable_end="B")),
> ).filter(Q(aterm__gte=1, bterm=0) | Q(aterm=0, bterm__gte=1) | Q(aterm=0, bterm=0))
> ```

Looks like that works, and doesn't have the problems I was coming up against.

File: scripts/find_orphaned_cables.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from django.db.models import Q
```
Comment (candlerb) [inline scripts/find_orphaned_cables.py:1]: This needs to be:

```
from django.db.models import Count, Q
```

otherwsie you get

> NameError: name 'Count' is not defined

File: scripts/find_orphaned_cables.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    def run(self, data, commit):
+        cables = Cable.objects.annotate(
+            aterm=Count('terminations', filter=Q(terminations__cable_end="A")),
+            bterm=Count('terminations', filter=Q(terminations__cable_end="B")),
+        ).filter(Q(aterm__gte=1, bterm=0) | Q(aterm=0, bterm__gte=1) | Q(aterm=0, bterm=0))
+        self.log_info(f'Found {cables.count()} problematic cables in DB')
+        for cable in cables:
+            if cable.aterm == 0 and cable.bterm > 0:
+                self.log_warning(f'{cable.pk}: {cable} is missing \'A\' side termination')
```
[the remark this answers is not in the document]
Comment (DanSheps) [reply scripts/find_orphaned_cables.py:19]: Thanks Brian,

Yeah, I will fix that up.

FYI, there is a button to the left of the "H" in the formatting bar that will let you suggest a replacement.

File: scripts/find_orphaned_cables.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+class BrokenCableTerminations(Script):
+    name = f'Find (partially) orphaned cables'
+    description = f'Find cable terminations misising either the A or B termination'
+
+    def run(self, data, commit):
+        cables = Cable.objects.annotate(
+            aterm=Count('terminations', filter=Q(terminations__cable_end="A")),
+            bterm=Count('terminations', filter=Q(terminations__cable_end="B")),
+        ).filter(Q(aterm__gte=1, bterm=0) | Q(aterm=0, bterm__gte=1) | Q(aterm=0, bterm=0))
```
[the remark this answers is not in the document]
Comment (DanSheps) [reply scripts/find_orphaned_cables.py:15]:
```suggestion
        ).filter(Q(bterm=0) | Q(aterm=0) )
```

File: scripts/find_orphaned_cables.py (hunk 1 above)
Comment (DanSheps) [reply scripts/find_orphaned_cables.py:1]:
```suggestion
from django.db.models import Count, Q
```

File: scripts/find_orphaned_cables.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (DanSheps) [reply scripts/find_orphaned_cables.py:19]:
```suggestion
            if cable.aterm == 0 and cable.bterm > 0:
                self.log_warning(f'[{cable}](/dcim/cables/{cable.pk}/) is missing \'A\' side termination')
            elif cable.aterm > 0 and cable.bterm == 0:
                self.log_warning(f'[{cable}](/dcim/cables/{cable.pk}/) is missing \'B\' side termination')
            else:
                self.log_warning(f'[{cable}](/dcim/cables/{cable.pk}/) is orphaned')
```

File: scripts/find_orphaned_cables.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        self.log_info(f'Found {cables.count()} problematic cables in DB')
+        for cable in cables:
+            if cable.aterm == 0 and cable.bterm > 0:
+                self.log_warning(f'{cable.pk}: {cable} is missing \'A\' side termination')
+            elif cable.aterm > 0 and cable.bterm == 0:
+                self.log_warning(f'{cable.pk}: {cable} is missing \'B\' side termination')
+            else:
+                self.log_warning(f'{cable.pk}: {cable} is orphaned')
```
Comment (DanSheps) [inline scripts/find_orphaned_cables.py:23]:
```suggestion
```

File: scripts/find_orphaned_cables.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    description = f'Find cable terminations misising either the A or B termination'
+
+    def run(self, data, commit):
+        cables = Cable.objects.annotate(
+            aterm=Count('terminations', filter=Q(terminations__cable_end="A")),
+            bterm=Count('terminations', filter=Q(terminations__cable_end="B")),
+        ).filter(Q(aterm__gte=1, bterm=0) | Q(aterm=0, bterm__gte=1) | Q(aterm=0, bterm=0))
+        self.log_info(f'Found {cables.count()} problematic cables in DB')
+        for cable in cables:
+            if cable.aterm == 0 and cable.bterm > 0:
```
Comment (DanSheps) [inline scripts/find_orphaned_cables.py:18]:
```suggestion
```

**Review by candlerb — APPROVED** [review]
LGTM!

**Pull request merged**